CI run against latest deps is failing

See https://github.com/matrix-org/synapse/actions/runs/2407332978

```
 synapse/config/_util.py:59: error: Subclass of "str" and "int" cannot exist: would have incompatible method signatures  [unreachable]
synapse/config/_util.py:60: error: Statement is unreachable  [unreachable]
```

https://github.com/matrix-org/synapse/blob/4b965c862dc66c0da5d3240add70e9b5f0aa720b/synapse/config/_util.py#L43-L63

I think the underlying change was the recent types-jsonschema release, see https://github.com/python/typeshed/pull/7950. The stubs now [say](https://github.com/sirosen/typeshed/blob/8ddc5934d3e9d06ef24bb9ce0f0d27a783f09237/stubs/jsonschema/jsonschema/exceptions.pyi#L41-L42) that `e.absolute_path` is a `Sequence[str]`.

--- a/changelog.d/12912.misc
+++ b/changelog.d/12912.misc
@@ -1,0 +1,1 @@
+Bump types-jsonschema from 4.4.1 to 4.4.6.

--- a/synapse/events/validator.py
+++ b/synapse/events/validator.py
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import collections.abc
-from typing import Iterable, Type, Union
+from typing import Iterable, Type, Union, cast
 
 import jsonschema
 
@@ -103,7 +103,12 @@ class EventValidator:
             except jsonschema.ValidationError as e:
                 if e.path:
                     # example: "users_default": '0' is not of type 'integer'
-                    message = '"' + e.path[-1] + '": ' + e.message  # noqa: B306
+                    # cast safety: path entries can be integers, if we fail to validate
+                    # items in an array. However the POWER_LEVELS_SCHEMA doesn't expect
+                    # to see any arrays.
+                    message = (
+                        '"' + cast(str, e.path[-1]) + '": ' + e.message  # noqa: B306
+                    )
                     # jsonschema.ValidationError.message is a valid attribute
                 else:
                     # example: '0' is not of type 'integer'

